### PR TITLE
Notification component vanilla extract rewrite [old]

### PR DESCRIPTION
**Pull request by bearni95**
## Reason:

Closes ["Recreate the Notifications Component"](https://app.asana.com/0/1204649083736950/1204713993965158/)

## Changes made (preferably with images/screenshots):
- [x] Rewrite the core of the notification component using `vanilla-extract`
- [x] Pass down "footer" as parameter (blue)
- [ ] Pass down "header" as parameter (red)
img
- [ ] Support for notification-icon and close-icon
 
## Check off the following:

- [ ] I have reviewed my changes and run the appropriate tests.
- [ ] I have have run `rush change` to add the appropriate change logs.
- [ ] I have added/edited docs.
- [ ] I have added tutorials.
- [ ] I have double checked and DEFINITELY added docs.

### CODE REVIEW

File: common/config/rush/pnpm-lock.yaml [hunk 1]
```diff
@@ -10851,7 +10852,6 @@ packages:
 
   /classnames/2.3.2:
     resolution: {integrity: sha512-CSbhY4cFEJRe6/GQzIk5qXZ4Jeg5pcsP7b5peFSDpffpe1cqjASH/n9UTjBwOp6XpMSTwQ8Za2K5V02ueA7Tmw==}
-    dev: false
```
Comment (eileenmguo) [inline common/config/rush/pnpm-lock.yaml:10854]: can you run `rush update` again and see if this goes away? I assume this is left over from when classnames was devDependency

File: packages/apps/kadena-docs/src/pages/docs/contribute/docs.mdx [hunk 2]
```diff
@@ -0,0 +1,221 @@
+---
```
Comment (eileenmguo) [inline packages/apps/kadena-docs/src/pages/docs/contribute/docs.mdx:1]: I feel like this shouldn't be part of this PR?

File: packages/libs/react-components/src/components/Input/Input.tsx [hunk 3]
```diff
@@ -16,8 +16,8 @@ export interface IInputProps
   > {
   as?: 'input';
   leadingText?: string;
-  leftPanel?: typeof SystemIcons[keyof typeof SystemIcons];
-  rightPanel?: typeof SystemIcons[keyof typeof SystemIcons];
+  leftPanel?: (typeof SystemIcons)[keyof typeof SystemIcons];
```
Comment (eileenmguo) [inline packages/libs/react-components/src/components/Input/Input.tsx:19]: Comment left in other PR as well: Do you know what is causing this formatting change? Maybe it's a global prettier file? The change isn't semantically wrong, but I've noticed when we've merged this change in the past, it gets changed back by other PRs so we want to avoid these formatting inconsistencies if possible

File: packages/libs/react-ui/package.json [hunk 4]
```diff
@@ -71,6 +71,7 @@
     "@types/node": "^16.0.0",
     "@types/react": "^18.0.30",
     "@vanilla-extract/webpack-plugin": "~2.2.0",
+    "classnames": "^2.3.1",
```
Comment (eileenmguo) [inline packages/libs/react-ui/package.json:74]: We should remove this from devDependencies since it needs to be a normal dependency

File: packages/libs/react-ui/src/components/Icons/IconWrapper.css.ts [hunk 5]
```diff
@@ -5,6 +5,7 @@ import { style, styleVariants } from '@vanilla-extract/css';
 export const iconContainer = style([
   sprinkles({
     color: 'foreground',
+    display: 'flex',
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Icons/IconWrapper.css.ts:8]: I removed this intentionally since the svg should be 100% width and height. Did you find an instance in which this is necessary?

File: packages/libs/react-ui/src/components/Notification/Notification.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+            Notification text to inform users about the event that occurred!
+          </NotificationBody>
+          <NotificationFooter>
+            <Button title="Action label" icon={SystemIcons.TrailingIcon}>
+              Action label
+            </Button>
+          </NotificationFooter>
+        </Notification>
+      </>
+    );*/
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Notification/Notification.stories.tsx:164]: cleanup?

File: packages/libs/react-ui/src/components/Notification/Notification.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Notification/Notification.stories.tsx:107]: cleanup?

File: packages/libs/react-ui/src/components/Notification/Notification.css.ts [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+const contrastColor = createVar(),
+  surfaceColor = createVar();
+
+export const containerClass = style([
+  sprinkles({
+    borderWidth: 'md',
+    display: 'flex',
+    justifyContent: 'flex-start',
+    alignItems: 'flex-start',
+    //paddingX: 'md',
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Notification/Notification.css.ts:20]: cleanup?

File: packages/libs/react-ui/src/components/Notification/Notification.css.ts [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+    width: 'max-content',
+    height: 'min-content',
+  }),
+  {
+    border: `1px solid ${fallbackVar(contrastColor, vars.colors.neutral6)}`,
+    color: fallbackVar(contrastColor, vars.colors.neutral6),
+    backgroundColor: fallbackVar(surfaceColor, vars.colors.neutral2),
+    borderLeftWidth: vars.sizes['1'],
+    position: 'relative',
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Notification/Notification.css.ts:29]: can this be a sprinkle?

File: packages/libs/react-ui/src/components/Notification/Notification.css.ts [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+      {
+        vars: {
+          [contrastColor]:
+            color === 'default'
+              ? vars.colors.neutral6
+              : vars.colors[`${color as ColorType}Contrast`],
+          [surfaceColor]:
+            color === 'default'
+              ? vars.colors.neutral2
+              : vars.colors[`${color as ColorType}Surface`],
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Notification/Notification.css.ts:95]: We generally want to avoid type casting so it would be better to define the type for the record or in the function signature

File: packages/libs/react-ui/src/components/Notification/Notification.css.ts [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+export const iconContainerClass = style([
+  sprinkles({
+    marginX: 'md',
+    marginTop: 'sm',
+    display: 'flex',
+    right: 0,
+  }),
+  {
+    position: 'relative',
+    // marginLeft:'auto'
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Notification/Notification.css.ts:111]: cleanup/move to sprinkles?

File: packages/libs/react-ui/src/components/Notification/Notification.css.ts [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+  },
+]);
+
+export const iconContainerFullWidthClass = style([
+  sprinkles({
+    marginTop: 'sm',
+    marginRight: 'md',
+  }),
+  { marginLeft: 'auto' },
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Notification/Notification.css.ts:120]: is this necessary? If so, do you think we should move this to sprinkles?

File: packages/libs/react-ui/src/components/Notification/Notification.css.ts [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+  }),
+  {
+    position: 'absolute',
+    right: '100%',
+  },
+]);
+
+globalStyle(`${headerContainerClass} > span`, {
+  marginLeft: vars.sizes.md,
+});
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Notification/Notification.css.ts:141]: Is there is a reason why you applied this style as a globalStyle? Maybe applying a paddingLeft on the header container class would suffice?

File: packages/libs/react-ui/src/components/Notification/Notification.stories.tsx [hunk 14]
```diff
@@ -0,0 +1,187 @@
+import { Button, SystemIcon } from './../../';
+import { INotificationProps, Notification } from './Notification';
+import { colorVariants } from './Notification.css';
+
+import type { Meta, StoryObj } from '@storybook/react';
+import React from 'react';
+
+const meta: Meta<
+  {
+    //selectIcon: string;
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Notification/Notification.stories.tsx:10]: cleanup?

File: packages/libs/react-ui/src/components/Notification/Notification.stories.tsx [hunk 15]
```diff
@@ -0,0 +1,187 @@
...
+        type: 'boolean',
+      },
+    },
+  },
+};
+
+export default meta;
+type Story = StoryObj<
+  {
+    //selectIcon: keyof typeof SystemIcons;
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Notification/Notification.stories.tsx:57]: cleanup?

File: packages/libs/react-ui/src/components/Notification/Notification.tsx [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+  icon?: (typeof SystemIcon)[keyof typeof SystemIcon];
+  title: string;
+  children?: React.ReactNode;
+  displayCloseButton?: boolean;
+  expand?: boolean;
+  color?: keyof typeof colorVariants;
+  simple?: boolean;
+
+  headerIcons?: React.ReactNode;
+  footer?: React.ReactNode;
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Notification/Notification.tsx:29]: Based on our previous discussion regarding subcomponents, I believe these shouldn't be passed as props, but rather have Notification sub components that people can use to compose a notification.

**Comment by bearni95** [comment]
Closing in favor of [this PR](https://github.com/kadena-community/kadena.js/pull/398), which contains all the changes without the weird, extra code
